### DISCUSSION TITLE
Activation problem of helper with Firefox

### DISCUSSION THREAD

**Discussion by deanWpennington** [aclap-dev/video-downloadhelper#1100] [Help and Q&A] [answered]
### Your question

I have latest Firefox and a valid purchased activation for the co-op. It does download but puts a QR on images. **When I go to the helpers settings it says "License not set" when I paste in the activation is takes it, and the fill in space returns blank and it still says "License not set"**  Knowing me I'm sure I did something wrong, but I don't know what? I got here by trying a BETA version then later uninstalled all, but until that it was all working. This is still the same PC it all worked on, reinstalling all gets this default condition in **bold** above. I could use some help. Companion App installed (v2.0.19) 

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\Dean\\Desktop"
        }
    },
    "view_options": {
        "all_tabs": true,
        "low_quality": true,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": true
    },
    "use_sidebar": false,
    "open_count_store": 10,
    "successfull_dl": 6
}

I did not see a problem like this one, and I have no insight if a solution to a differing problem would solve mine as well. but then I got in this fix by trying the beta/

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

Take this as the wild-ass guess it is.  I'm going to guess you have some kind of leftover from a previous install lurking in your registry.  When you uninstall the CoApp, you should not do it via the Windows function for uninstalling apps.  You should use Revo Uninstaller instead.  This is a piece of brilliant freeware.  Yes, truly free.  It's one of those deals where the guy charges corporate users to use it but home users get a break.  I've been using it for several years, originally on W7 & now on W11.

So I would try again.  Get Revo.  Do a web search to find it.  Install it.  You should also prepare for reinstalling the VDH browser extension, which is in my advice below.  You should download the xpi file to your system.  Since you'll be starting over fresh, you ought to get the beta version of VDH.  You say you installed it at some point.  I believe the beta did not cause your problem.  You should not look unkindly upon the beta.  I've been running betas of VDH for years.  To get the xpi on your system, find the beta in the Wiki again.  But instead of just clicking the button, pop up its context menu.  Then execute the `Save Link As...` function.  This will give you the chance to navigate to a direectory on your system & save the xpi file where you can find it below.

Uninstall the VDH browser extension using the usual tools inside Firefox.  Then close all browser windows.  Then Revo uninstall the CoApp.  This occurs in more than one phase.  First, Revo just runs the CoApp's identified uninstaller.  Then Revo analyzes both the registry & your disk space for leftovers.  It prompts you before it removes anything.  Check those prompts carefully & make sure it's proposing to delete things you agree Revo should be deleting.  I can't recall a time when I ever disagreed with what it has proposed, but I have ALWAYS double checked what it says it will delete.  Always be careful with this sort of tool.  It modifies your registry.  Be careful with it.

Once that's done, reopen Firefox.  Do a Ctrl+o to open a file.  Navigate to the xpi file you got earlier & execute it.  This will intall the VDH beta.  Close all browser windows.

Execute the CoApp installer.

Open Firefox & see if you can enter your license now.  Post again to report your experience doing all this.

**Reply by GrampaWildWilly** [reply]
Oh.  Meant to mention this.  When you paste in your license key, make sure you really are pasting in the whole thing.  Double check you are highlighting the whole thing with no extras.  When you paste, double check it again.  Fat fingering is my favorite passtime.  You shouldn't fall victim to it.  My license key is 32 characters.  I assume that's true for everybody.  Maybe not.  But I would think it is.

**Comment by paulrouget** [comment, marked as answer]
I know this issue! I fixed it in version 9.1 (no need to use beta). Update your addon :)
